lmos-runtime-graphql-service/src/main/kotlin/org/eclipse/lmos/runtime/graphql/service/inbound/subscription/ConversationSubscription.kt

```diff
@@ -10,31 +10,63 @@ import com.expediagroup.graphql.generator.annotations.GraphQLDescription
 import com.expediagroup.graphql.server.operations.Subscription
 import kotlinx.coroutines.coroutineScope
 import kotlinx.coroutines.flow.flow
+import org.eclipse.lmos.arc.api.AgentRequest
+import org.eclipse.lmos.arc.api.AgentResult
+import org.eclipse.lmos.arc.api.Message
 import org.eclipse.lmos.runtime.core.inbound.ConversationHandler
-import org.eclipse.lmos.runtime.core.model.Conversation
+import org.eclipse.lmos.runtime.core.model.*
 import org.springframework.stereotype.Component
 
 @Component
 class ConversationSubscription(private val conversationHandler: ConversationHandler) : Subscription {
     @GraphQLDescription("Processes the user input and returns the result")
-    suspend fun chat(
-        conversation: Conversation,
-        conversationId: String,
-        tenantId: String,
-        turnId: String,
+    suspend fun agent(
+        agentName: String? = null,
+        request: AgentRequest,
     ) = flow {
         coroutineScope {
```

Comment:
Did you try to reuse the flow which is returned by handleConversation and just map?
Or is a new coroutineScope needed?

```
conversationHandler.handleConversation(
                    conversation,
                    conversationId,
                    tenantId,
                    turnId,
                    conversation = request.toConversation(),
                    conversationId = request.conversationContext.conversationId,
                    tenantId = request.systemContext.first { it.key == "tenantId" }.value,
                    turnId = request.conversationContext.turnId ?: "unknown turnId",
                ).map()
```

Reply:
I asked myself the same question, but did not want to change this here, as it's (imho) orthogonal to the change I did (which is only about adapting the interface of the graphql endpoint).

My gut feeling is also that one might be able to avoid the `coroutineScope` - but I did not experiment with that yet.

I, personally, would prefer to leave it as is for this PR, but could take a look if this could be improved at a later point.